opengever/maintenance/scripts/export_dossier_structure_as_excel.py

```diff
@@ -27,12 +28,21 @@ def __init__(self, obj):
         refnum = reference_number.get_parent_numbers()
         self.depth = (len(refnum['repository']) +
                       len(refnum.get('dossier', [])))
+
+        self.url = self.get_url(obj)
+
         # we cannot use IReferenceNumberFormatter.sorter, as it does not sort
         # correctly for a mix of repository folders and dossiers. Instead we
         # make our own sorting key
         self.sorting_key = (map(int, refnum.get('repository', tuple())),
                             map(int, refnum.get('dossier', tuple())))
 
+    def get_url(self, obj):
```

Comment:
Getting the site-relative path of the object can be done using [`url_tool.getRelativeContentPath()`](https://github.com/zopefoundation/Products.CMFCore/blob/master/src/Products/CMFCore/interfaces/_tools.py#L1726-L1734):

```python
url_tool = api.portal.get_tool('portal_url')
(Pdb) self.context.getPhysicalPath()
('', 'fd', 'ordnungssystem', 'fuehrung', 'gemeinderecht', 'dossier-21')
(Pdb) url_tool.getRelativeContentPath(self.context)
('ordnungssystem', 'fuehrung', 'gemeinderecht', 'dossier-21')
```
Not that it matters a lot, since the [implementation](https://github.com/zopefoundation/Products.CMFCore/blob/master/src/Products/CMFCore/URLTool.py#L80-L86) of that  is the exact same 😉